Add "Copy as cURL" Feature

oleg-kuibar opened the pull request:
## Overview
Added the ability to copy GraphQL requests as cURL commands, making it easier for users to share and debug requests outside the extension.

Closes #118 

## Changes
1. Added new `CopyAsCurlButton` component
   - Extends functionality of existing CopyButton
   - Handles cURL command generation for network requests
   - Includes tests for component behavior

2. Added cURL generation helper
   - New `curlHelpers.ts` with `getChromeNetworkCurl` function
   - Generates cURL commands from network request data
   - Includes tests for different request scenarios

3. Updated NetworkDetails view
   - Added CopyAsCurlButton next to existing copy buttons
   - Passes network request data to the new component

4. Updated manifest.json
   - Added required permissions for network request handling
   - Updated content script configuration

## Testing
- Unit tests added for new components and helpers
- Manual testing of cURL command generation
- Verified cURL commands work when pasted in terminal

## Notes
- Maintains existing copy functionality
- Follows existing code patterns and styles
- All tests passing

## List of Remaining Issues ‼️ 
- [x] Headers still include pseudo-headers (`:authority`, `:method`, etc.) which Chrome's cURL doesn't include
- [x] Body escaping doesn't match Chrome's format exactly (Chrome uses `\u0021` for `!` and other unicode escapes)

---

https://github.com/user-attachments/assets/0b836414-461f-4c26-a45b-3652ead6f1d3

https://github.com/user-attachments/assets/20907521-72df-4d1c-823d-3188a3fde76f

warrenday commented:
Hey @oleg-kuibar is this one ready for review? Thanks for raising.

oleg-kuibar commented:
> Hey @oleg-kuibar is this one ready for review? Thanks for raising.

Hey @warrenday, the PR is not yet ready for review. I aim to resolve remaining issues this week. Thank you!

oleg-kuibar commented:
@warrenday ready for review, ty!

warrenday commented on scripts/set-manifest-content-script.js:
```diff
@@ -5,9 +5,15 @@ const manifestPath = "./build/manifest.json"
+  
+  // Ensure URL has scheme
+  const baseUrl = process.env.REACT_APP_SHARE_TARGET_URL || 'http://localhost:3000'
+  const url = baseUrl.startsWith('http') ? baseUrl : `http://${baseUrl}`
```
Might be better to default to https with these checks. As the REACT_APP_SHARE_TARGET_URL will always be a secure url.

oleg-kuibar commented:
> Hey @oleg-kuibar PR looks great, one query on the `http` check and a couple tests failed, otherwise no other changes. If your tests are passing locally without issue and only failing on CI, let me know and I can help debug.
> 
> I have enabled the test suite to run, so _should_ run on every push now.

I've addressed the comments. I also noticed that workflow requires approve before it will run

warrenday commented:
Still have a couple failing tests, are these passing locally for you? PR looks good, so can merge once tests are passing.

oleg-kuibar commented:
> Still have a couple failing tests, are these passing locally for you? PR looks good, so can merge once tests are passing.
They do pass. I'll try to clone and do clean install and address failing parts if they will show up, ty!

warrenday commented:
Great, tests now passing. I'll merge this and get a release out soon... thank you!

warrenday reviewed (approved):
Great work!
